h2o-py/tests/testdir_algos/automl/pyunit_automl_leaderboard.py

```diff
@@ -293,6 +293,29 @@ def test_AUTO_stopping_metric_with_custom_sorting_metric():
     check_model_property(non_se, 'stopping_metric', True, "RMSE")
 
 
+def test_custom_leaderboard():
+    print("Check custom leaderboard")
+    ds = prepare_data('binomial')
+    aml = H2OAutoML(project_name="py_aml_custom_lb_test",
```

Comment:
@seb-h2o I was thinking about whether we treat default columns as default extensions... Don't we want to sort based on extension columns?
```
aml = H2OAutoML(project_name="py_aml_custom_lb_test",
                    max_models=5, sort_metric = "training_time_ms",
                    seed=automl_seed)
get_leaderboard(aml, extensions=["training_time_ms"]).names == std_columns + ["training_time_ms"]
```
Maybe we can lazily keep sorting metric and only up on request return or fail depending on whether column exists in the leaderboard frame?

Reply:
leaderboard is represented as a `H2OFrame` on client side, user can do the sorting by himself then: no point to complicate the backend logic for a functionality that you get for free on client side.

Reply:
@seb-h2o maybe, but we already have `sort_metric = "training_time_ms"` functionality for H2OAutoML. Just to follow existing contract I would expect to support this for extensions as well. Or we should at least for ourselves distinguish original metrics and extensions.

Reply:
I called those "extensions" to make it clear that they are distinct from "metrics", so I don't expect `sort_metric` to apply here.
Besides, it would complicate the implementation a lot given how sorting is done today on backend, for almost zero benefit.
If people really want it,  we can do it later, but currently it doesn't make sense imo.

Is it really difficult to do:
```
h2o.automl.get_leaderboard(aml, 'ALL').sort(by='training_time_ms')
```
I think duplicating features is the best way to create inconsistent APIs.

Reply:
It was not about duplicating features but about making original one to be working with new extensions ( which could be seen as columns we might want to sort by as well).  The only point I wanted to communicate is that we already have this feature and to support it we would want to support sorting based on extension columns. If we don't wan't to support we need to deprecate existing `sort_metric` and suggest to use sorting from Frames.  But as you have said those are "extensions"... and we distinguish between them... then we are fine. Thanks for clarifying!

Reply:
It's true that we have `sort_metric` in AutoML and then `sort_by` in the get_grid code.   But since it's metric specific (and our new columns are not metrics) it seems less relevant.  It's not a pervasive convention either. I agree that people can sort a data frame on their own... the only non-trivial part is that the leaderboard is an H2OFrame and so users need to know how to sort H2OFrames, which unfortunately is not the same way that you can sort R data.frames or Python Pandas DataFrames.  So it's more awkward than it should be. http://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-munging/sortcolumn.html

Reply:
@ledell this can be shown through examples, let me add a line for sorting in the `get_leaderboard` Py example.
Also, h2o frames can easily be converted to pandas df.
We may want to have a detailed example in the online doc.

Reply:
@seb-h2o yeah we just need more documentation